ldi-api/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/types/LdiAdapter.java

```diff
@@ -0,0 +1,16 @@
+package be.vlaanderen.informatievlaanderen.ldes.ldi.types;
+
+import org.apache.jena.rdf.model.Model;
+
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import static be.vlaanderen.informatievlaanderen.ldes.ldi.types.LdiAdapter.InputObject;
+
+public interface LdiAdapter extends LdiComponent, Function<InputObject, Stream<Model>> {
+	record InputObject(String content, String contentType) {
```

Comment:
Not sure if I would couple this so thightly to the adapter interface. A simple 'Content' record as minimal standalone could be useful elsewhere.

Reply:
I agree on the naming. However, since this object has no usage apart from the Ldi Adapter, I fear the name Content, or whatever name we try to give it, will be too vague and will easily be overlooked.
Providing an internal class provides more context to the Adapter too